test/gemm/gemm_kernel_base_impl.hpp

```diff
@@ -338,6 +338,43 @@ namespace rocwmma
         return ldsUsage() <= DeviceInfo::instance()->sharedMemSize();
     }
 
+    template <uint32_t BlockM,
+              uint32_t BlockN,
+              uint32_t BlockK,
+              typename InputT,
+              typename OutputT,
+              typename ComputeT,
+              typename LayoutA,
+              typename LayoutB,
+              typename LayoutC,
+              typename LayoutD>
+    bool GemmKernelBase<BlockM,
+                        BlockN,
+                        BlockK,
+                        InputT,
+                        OutputT,
+                        ComputeT,
+                        LayoutA,
+                        LayoutB,
+                        LayoutC,
+                        LayoutD>::checkTypes() const
+    {
+        // check if hardwares support data type
+        auto deviceArch = DeviceInfo::instance()->getGcnArch();
```

Comment:
This shouldn't be needed. The gemm_predicates_base.hpp checks device info through the dispatcher in gemm_kernel_base_dispatch_impl.hpp.